_posts/2022-05-18-My-OTP-25-highlights.md

```diff
@@ -0,0 +1,751 @@
+---
+layout: post
+title: Erlang/OTP 25 Highlights
+tags: erlang otp 25 release
+author: Kenneth Lundin
+---
+
+OTP 25 is finally here. This post will introduce the new features that I am most excited about.
+
+Erlang/OTP 25 includes contributions from ??+ external contributors totalling
+??+ commits, ??+ PRs.
+
+You can download the readme describing all the changes here: [Erlang/OTP 25 Readme].
+Or, as always, look at the release notes of the application you are interested in.
+For instance here: [Erlang/OTP 25 - Erts Release Notes - Version 13.0].
+
+This years highlights are:
+* [New functions in the `maps`and `lists` modules](#new-functions-in-the-mapsand-lists-modules)
+* [Selectable features and the new `maybe_expr` feature](#selectable-features-and-the-new-maybe_expr-feature)
+* Compiler News
+* Crypto and OpenSSL 3.0
+* Dialyzer
+* [Improvements of the JIT](#improvements-of-the-jit)
+* ETS-tables with adaptive support for write concurrency
+* Relocatable installation directory for Erlang
+* New option `short` to the functions `erlang:float_to_list/2` and `erlang:float_to_binary/2` 
+* The new module `peer` supersedes the slave module
+* global will now by default prevent overlapping partitions
+* gen_server, gen_statem and gen_event has got a new format_status/1 callback.
+* The `timer` module has been modernized and made more efficient
+* CA-certificates can be fetched from the OS standard place 
+
+
+# New functions in the `maps`and `lists` modules
+
+Triggered by suggestions from the users we have introduced new functions in the `maps` and `lists` modules in `stdlib.
+
+## `maps:groups_from_list/2,3`
+
+For short we can say that this function take a list of elements and group them. The result is a map `#{Group1 => [Group1Elements], GroupN => [GroupNElements]}`.
+
+Let us look at some examples from the shell:
+
+```erlang
+> maps:groups_from_list(fun(X) -> X rem 2 end, [1,2,3]).
+#{0 => [2], 1 => [1, 3]}
+```
+The provided fun calculates X rem 2 for every element X in the input list and then group the elements in a map with the result of X rem 2 as key and the corresponding elements as a list for that key.
+
+```erlang
+> maps:groups_from_list(fun erlang:length/1, ["ant", "buffalo", "cat", "dingo"]).
+#{3 => ["ant", "cat"], 5 => ["dingo"], 7 => ["buffalo"]}
+```
+
+In the example above the strings in the input list are grouped into a map based on their length.
+
+## `maps:groups_from_list/3`
+
+There is also a variant of groups_from_list with an additional fun by which the values can be converted or changed before they are put into their groups.
```

Comment:
What's the difference between 'converted' and 'changed' ?

Reply:
Unnecessary to have both. Removed changed.